### PR TITLE
docs: HowTo IRouteNotifier

### LINKED ISSUE #1679: [Navigation][Documentation] Write advanced topic with `IRouteNotifier`, `NavigationResponse`, `INavigator`
> @lukeblevins can we create an advanced topic that looks at using the IRouteNotifier and the NavigationResponse to access the INavigator. An example where this would be useful is opening a dialog where you might want to cancel the dialog via the INavigator instance

_Originally posted by @nickrandolph in https://github.com/unoplatform/uno.extensions/issues/1678#issuecomment-1645238935_

### PR DESCRIPTION
**Pull request by eriklimakc**
GitHub Issue (If applicable): closes #1679

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## What is the new behavior?

Add docs to the Advanced section of Navigation regarding `IRouteNotifier` and how to access `INavigator` through the `RouteChanged` event.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested with docfx
- [X] Associated with an issue (GitHub or internal)

### CODE REVIEW

**Comment by agneszitte** [comment]
@eriklimakc what is the status for this PR please

**Comment by eriklimakc** [comment]
> @eriklimakc what is the status of this PR please ?

@agneszitte I still need to add usage examples for IRouteNotifier, NavigationResponse, INavigator.

File: doc/Learn/Navigation/Advanced/HowTo-IRouteNotifier.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+The `IRouteNotifier` interface allows you to track and respond to route changes through the `RouteChanged` event. This guide will show you how to use `IRouteNotifier` to handle route changes and improve your app's navigation.
+
+## Step-by-steps
+
+> [!IMPORTANT]
+> This guide assumes you used the template wizard or `dotnet new unoapp` to create your solution. If not, it is recommended that you follow the [Creating an application with Uno.Extensions article](xref:Uno.Extensions.HowToGettingStarted) for creating an application from the template.
+
+### How to Use `IRouteNotifier` to Monitor Route Changes
+
+The `IRouteNotifier` provides an effective way to monitor and respond to route changes within your application. To begin using `IRouteNotifier` Ensure your class has access to an instance of an `IRouteNotifier` implementation. Add a parameter of type `IRouteNotifier` to the constructor of your class where you want to monitor route changes.
```
Comment (kazo0) [inline doc/Learn/Navigation/Advanced/HowTo-IRouteNotifier.md:15]:
```suggestion
The `IRouteNotifier` provides an effective way to monitor and respond to route changes within your application. To begin using `IRouteNotifier`, ensure your class has access to an instance of an `IRouteNotifier` implementation. Add a parameter of type `IRouteNotifier` to the constructor of your class where you want to monitor route changes.
```

**Review by kazo0 — APPROVED** [review]

**Comment by eriklimakc** [comment]
@Mergifyio backport release/stable/5.0

**Pull request merged**